### DISCUSSION TITLE
Any way to show first frame or recorded blob URL,

### DISCUSSION THREAD

**Discussion by raviprakash007** [videojs/video.js#7592] [Ideas]
Hi Team, 
Is there any way to set poster source dynamically for video recordings done by WebRTC.  We get recorded video as blob and I want to show first frame of recorded blob as poster image.

**Comment by SomeDevWeb** [comment]
Hi. 
This looks like a great player and I intend to use it for a project.
Very similar situation here. I have a case where I can load the video URL from the filesystem (via PHP), but no way to set the poster URL, so I need a solution to dynamically load a frame from within the video as the poster.
I tried the solution with "#t=0.1" but it doesn't work. 
Another note is that the first frames are black as the video starts with a slow fade from dark, so it would need to be a custom frame. 
Thank you for any feedback!

**Comment by gkatsev** [comment]
Your best bet is to generate a frame in another method and supply the image as the poster attribute to the video element.

**Reply by SomeDevWeb** [reply]
Thanks for the quick reply. Well, that's exactly my problem, HOW to generate that frame? :)

**Reply by gkatsev** [reply]
You can do it via ffmpeg, for example https://stackoverflow.com/a/35026487

**Reply by SomeDevWeb** [reply]
OMG, I went throug a whole saga for a few hours with the hosting team which in the end said they can't install anything like that. 
So firstly, thanks a lot for your information and help which **is very valid in itself**, but I don't understand how some people can suggest something when they know full well that most of the cases that would not be valid...

**Reply by SomeDevWeb** [reply]
Secondly, I just tried the simple `<video>` tag WITH the "#t=0.1" and that works very well, so again I don't get how can a super-advanced player lack (or remove, I don't know) this feature of the most basic player.... 
Again, I KNOW how much effort goes into these FOSS project, so MY GREATEST APPRECIATION, but I'm so pissed off at things not working everywhere. There's gotta be a way to allow at least tha basic feature of the default playes, isn't it???

**Reply by gkatsev** [reply]
are you playing an mp4?

**Reply by SomeDevWeb** [reply]
Yes

**Reply by gkatsev** [reply]
I probably was overthinking things and assumed you had HLS or some other more complex scenario like the original post (with WebRTC). Let me check how things are.

**Reply by gkatsev** [reply]
I'm actually failing to see the media fragment show any poster in chrome, firefox, or safari with just a video element.

**Reply by gkatsev** [reply]
Oops, I was accidentally testing with `#0.1` instead of `#t=0.1`.

**Reply by gkatsev** [reply]
In my quick test using a simple videojs player, I can see the media fragment working with and without video.js. For mp4s, we just pass the url through to the video element and let the browser handle things.
If you have a live minimal example, I can try and take a look.
